Yagna uninstall instructions (#256)
golemfactory/golem-docs: src/pages/docs/providers/provider-uninstall.md

---

> title: How to uninstall Golem Provider?

Reviewer: ```suggestion
title: How to uninstall the Golem Provider
```

description: Step-by-step guide to removing Yagna components, configs, and data for Providers.
type: Instructions
---

> # Golem Provider uninstallation guide

Reviewer: ```suggestion
# How to uninstall the Golem Provider
```

{% alert level="warning" %}
Before removing Yagna, please make sure you have backed up your private key. You can find the instructions [here](/docs/providers/wallet/backup).
{% /alert %}

## Quick path

If you installed the provider using our quick-installer, execute the following commands to remove all associated components:

```bash
rm ~/.local/bin/yagna
rm ~/.local/bin/golemsp
rm ~/.local/bin/ya-provider
rm -rf ~/.local/share/yagna
rm -rf ~/.local/share/ya-installer
rm -rf ~/.local/share/ya-provider
rm -rf ~/.local/lib/yagna
```

## Detailed instructions for custom installations

### Removing binaries

Yagna binaries are typically located at: `~/.local/bin/`. Run the following command to remove all of the components.

```bash
rm ~/.local/bin/yagna && rm ~/.local/bin/golemsp && rm ~/.local/bin/ya-provider
```

> ### Removing Yagna binaries

Reviewer: The title is incorrect. Its not the binaries in the datadir.

Yagna daemon configuration and persistent files are stored in the location defined by the `YAGNA_DATADIR` environment variable. If it was not set, then the default location for Yagna data is: `~/.local/share/yagna`.

To verify if you use a custom data location, execute the following command:

```bash
echo $YAGNA_DATADIR
```

To remove the data in the default location, execute the command:

```bash
rm -r ~/.local/share/yagna
```

## Removing Yagna installer files

> To remove the data in the default location, execute the command:

Reviewer: To me it's not clear what these files cover. I don't know whats in the directory and what its used for

Author: binaries are there

```bash
rm -r ~/.local/share/ya-installer
```

## Removing Provider data

The configuration files, cached images, logs, and ExeUnit data directories for the provider agent are all stored in the directory specified by the `$DATADIR` environment variable. Should this variable be unset, the system defaults to using the following location: `~/.local/share/ya-provider`.

To verify if you're using a custom data location, execute the command:

```bash
echo $DATADIR
```

To remove the data in the default location, execute the command:

```bash
rm -r ~/.local/share/ya-provider
```

## Removing the Runtimes directory

> The runtimes directory is defined by the `$EXE_UNIT_PATH` environment variable. If this variable is not defined the default location is: `~/.local/lib/yagna`.

Reviewer: ```suggestion
The runtimes directory is defined by the `EXE_UNIT_PATH` environment variable. If this variable is not defined the default location is: `~/.local/lib/yagna`.
```

To verify if you use a custom data location, execute the following command:

```bash
echo $EXE_UNIT_PATH
```

To remove the data in the default location, execute the command:

```bash
rm -r ~/.local/lib/yagna
```

{% docnavigation title="See also" %}

> - If you are looking for intructions for Requestors see [here](/docs/creators/javascript/examples/tools/requestor-uninstall).

Reviewer: ```suggestion
- If you're looking for intructions on how to uninstall the requestor, [click here](/docs/creators/javascript/examples/tools/requestor-uninstall).
```

---
title: How to remove Yagna?
description: Step-by-step guide to removing Yagna components, configs, and data for Providers.
type: noindex
---

# Golem Provider uninstallation guide

{% alert level="warning" %}

> Before removing Yagna please make sure you have backup-up your private key to the address where you received GLMs. You can find instructions [here](/docs/providers/wallet/backup).

Reviewer: ```suggestion
Before removing Yagna, please make sure you have backed up your private key. You can find the instructions [here](/docs/providers/wallet/backup).
```

{% /alert %}

## Quick path

> If you used default settings for installation, this set of commands should be sufficient:

Reviewer: ```suggestion
If you installed the provider using our quick-installer, execute the following commands to remove all associated components:
```

```bash
rm ~/.local/bin/yagna
rm ~/.local/bin/golemsp
rm ~/.local/bin/ya-provider

> rm -fr ~/.local/share/yagna

Reviewer: ```suggestion
rm -rf ~/.local/share/yagna
```

> rm -fr ~/.local/share/ya-installer

Reviewer: ```suggestion
rm -rf ~/.local/share/ya-installer
```

> rm -fr ~/.local/share/ya-provider

Reviewer: ```suggestion
rm -rf ~/.local/share/ya-provider
```

> rm -fr ~/.local/lib/yagna

Reviewer: ```suggestion
rm -rf ~/.local/lib/yagna
```

```

## Detailed instructions for custom installations

### Removing binaries

The default Yagna binaries (actually link to binaries) location is: `~/.local/bin/yagna`.

```bash

> rm ~/.local/bin/yagna

Reviewer: ```suggestion
rm ~/.local/bin/yagna && rm ~/.local/bin/golemsp && rm ~/.local/bin/ya-provider
```

Let's just make it simple with a one-liner

> rm ~/.local/bin/golemsp

Reviewer: ```suggestion

```
